es6/tests/base.js

```diff
@@ -916,3 +920,65 @@ describe("Serialization", function() {
 		JSON.stringify(doc);
 	});
 });
+
+describe("Docxtemplater v4 tests", function() {
+	it("should work with new constructor", function() {
+		let xmlDocuments;
+
+		const module = {
+			requiredAPIVersion: "3.0.0",
+			optionsTransformer(options, docxtemplater) {
+				const relsFiles = docxtemplater.zip
+					.file(/document.xml.rels/)
+					.map(file => file.name);
+				options.xmlFileNames = options.xmlFileNames.concat(relsFiles);
+				return options;
+			},
+			set(options) {
+				if (options.xmlDocuments) {
+					xmlDocuments = options.xmlDocuments;
+				}
+			},
+		};
+
+		const moduleV4Templater = {
+			requiredAPIVersion: "3.0.0",
+			optionsTransformer(options, docxtemplater) {
+				const relsFiles = docxtemplater.zip
+					.file(/document.xml.rels/)
+					.map(file => file.name);
+				options.xmlFileNames = options.xmlFileNames.concat(relsFiles);
+				return options;
+			},
+			set(options) {
+				if (options.xmlDocuments) {
+					xmlDocuments = options.xmlDocuments;
+				}
+			},
+		};
+
+		const doc = createDoc("tag-example.docx");
+		doc.attachModule(module);
+		doc.compile();
+		doc.render();
+
+		// create new docxTemplater with new constructor
+		const docV4Templater = createDocV4(doc.getZip(), { modules: [moduleV4Templater] });
+		const xmlKeys = Object.keys(xmlDocuments);
+		expect(xmlKeys).to.deep.equal(["word/_rels/document.xml.rels"]);
+		const rels = xmlDocuments[
+			"word/_rels/document.xml.rels"
+		].getElementsByTagName("Relationship");
+		expect(rels.length).to.equal(10);
+		const str = xml2str(xmlDocuments["word/_rels/document.xml.rels"]);
+
+		if (isNode12()) {
```

Comment:
Why the isNode12 check ?

In general, everything should work on versions maintained by node (list is in .travis.yml)

Reply:
I could not figure out how to test the functioning of the module, So I used the code from `modules` test file.
I will figure out what this block of code is doing and change accordingly.

Reply:
To test the functioning of the module, I would do the following : 

```
let moduleCalled = false
module = {
   optionsTransformer(options) { moduleCalled = true; return options }
}
expect(moduleCalled).toBe(true);
```

I want to have the test as small as possible

Reply:
Awesome idea. I would do that. 👍
Still, I need to understand the functioning/lifecycle of modules better.